tools/explorer/pkg/escrow/escrow.go

```diff
@@ -147,10 +147,16 @@ func (e *Escrow) refundExpiredReservations() error {
 		return errors.Wrap(err, "failed to load active reservations from escrow")
 	}
 	for _, escrowInfo := range reservationEscrows {
-		if err := e.refundEscrow(escrowInfo); err != nil {
-			log.Error().Err(err).Msgf("failed to refund reservation escrow")
-			continue
+		log.Info().Int64("id", int64(escrowInfo.ReservationID)).Msg("expired escrow")
+
+		if escrowInfo.Paid { //only refund if the reservation has been paid before
```

Comment:
The `paid` flag is used by the escrow to identify reservations which have been fully funded. This means that if I create a reservation, pay half of it, and then do nothing, `paid` will still be false. This way we allow for incremental payment, or payment from multiple sources, ... So this check should not be here, as it means in the scenario described your partial payment will be lost to the escrow and not refunded. 

Also, it is technically fine to call the refund method multiple times on an escrow

Reply:
How do you recommend to handle this then ? 
Cause here we do refund any expired transaction even when it has not been funded. So we are giving up free money

Reply:
No, this handles expired escrows. With the change in the db method to load expired escrows, an expired escrowed is now considered as follows:

Any escrow, tied to a reservation who's `expiration_provision` field is set to less than the current timestamp (i.e. we can no longer provision it). Also, we check the `released` and `canceled` fields are not set, so we are sure we did not cancel the reservation, or the reservation is not successful yet.

This means that we should refund ALL of these reservations, even if they are successfully paid already. If we only refund fully funded reservations, we indeed now refund reservations which have the provisioning expired while deploying (when node is down or so). But now we won't refund partially paid reservations, so this money will be stuck. Likewise, if we only pay not fully funded reservations back to the client, something like a node being down can cause the payment to not be refunded and end up with the money stuck. So we need to refund in both cases. 

Because we filter out reservations which have been canceled previously, or released (deployed successfully and farmers paid), there is no harm in doing this in either case. Also, the refund just checks how much money is in the escrow *for this reservation* and refunds it. So we only give to the client what he actually send us in the first place (again, for this reservation).

TL;DR: the check can just go away and we can refund everything we pulled from db in this loop